File: LeetCode/007.Reverse Integer/reverse-integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
    let max = Math.pow(2, 31) - 1;
    let min = Math.pow(-2, 31);
    let symbol = x >= 0;
    let s = Math.abs(x).toString();
    let r = 0;
    for (let i = s.length - 1; i >= 0; i--) {
        r += s[i] * Math.pow(10, i);
    }
    r = symbol ? r : 0 - r;
    return r >= min && r <= max ? r : 0;
};

console.log(reverse(123));
console.log(reverse(-123));
console.log(reverse(120));

File: LeetCode/Go-Solutions/ProblemSet/0007.Reverse_Integer/reverse-integer.go
package _007_Reverse_Integer

import "math"

func reverse(x int) int {
	symbol := 1
	if x < 0 {
		symbol = -1
		x = 0 - x
	}

	r := 0
	for x > 0 {
		tmp := x % 10
		r = r*10 + tmp
		x = x / 10
	}

	r = symbol * r

	if r > math.MaxInt32 || r < math.MinInt32 {
		r = 0
	}
	return r
}
